src/test/unit/profile-data.test.js

```diff
@@ -589,7 +589,7 @@ describe('upgrades', function() {
     expect(serializedLhsAsObject).toEqual(serializedRhsAsObject);
   }
   const afterUpgradeReference = unserializeProfileOfArbitraryFormat(
-    require('../fixtures/upgrades/processed-7.json')
```

Comment:
Yeah, I think it's reasonable to create a new test for 7 -> 8 upgrade. So I would add a new file `processed-7a.json` and use that to test 7 -> 8 upgrade.

I'm not sure about the gecko v8 profile but I'll try what you described.